output/batch/table/formatter.go

```diff
@@ -0,0 +1,56 @@
+package table
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/olekukonko/tablewriter"
+
+	"github.com/cube2222/octosql"
+	"github.com/cube2222/octosql/execution"
+	"github.com/cube2222/octosql/output/batch"
+)
+
+func TableFormatter(rowLines bool) batch.TableFormatter {
+	return func(w io.Writer, records []*execution.Record, watermark time.Time, errToPrint error) error {
+		var fields []string
```

Comment:
This code appears more than once and it's like 12 lines of code - maybe create a function `getAllFields(records []*Record) []string`